src/otg/dataset/summary_statistics.py

```diff
@@ -122,14 +123,71 @@ def pvalue_filter(self: SummaryStatistics, pvalue: float) -> SummaryStatistics:
         )
         return SummaryStatistics(_df=df, _schema=self._schema)
 
-    def window_based_clumping(self: SummaryStatistics, distance: int) -> StudyLocus:
-        """Perform distance-based clumping.
+    def window_based_clumping_with_locus(
+        self: SummaryStatistics,
+        distance: int,
+        gwas_signiciance: float = 5e-8,
+        baseline_significance: float = 0.05,
+    ) -> StudyLocus:
+        """Generate study-locus from summary statistics by distance based clumping + collect locus.
 
         Args:
-            distance (int): Distance in base pairs
+            self (SummaryStatistics): _description_
+            distance (int): _description_
+            gwas_signiciance (float, optional): _description_. Defaults to 5e-8.
+            baseline_significance (float, optional): _description_. Defaults to 0.05.
 
         Returns:
-            StudyLocus: StudyLocus object
+            StudyLocus: _description_
         """
         # Calculate distance-based clumping:
-        return WindowBasedClumping.clump(self, distance)
+        return WindowBasedClumping.clump_with_locus(
+            self,
+            window_length=distance,
+            p_value_significance=gwas_signiciance,
+            p_value_baseline=baseline_significance,
+        )
+
+    def window_based_clumping(
+        self: SummaryStatistics, distance: int, gwas_signiciance: float = 5e-8
```

Comment:
At least at this level, we could make the `include_locus` a parameter of the function `window_based_clumping` function.

Reply:
You right, I'll change that.

Reply:
Let's do it in another PR